File: WorldFeed/Server/src/Services/Identity/WorldFeed.Identity.API/Application/Web/ITwitterAccessor.cs
namespace WorldFeed.Identity.API.Application.Web
{
    using System.Threading.Tasks;

    using WorldFeed.Identity.API.Application.Requesters;

    public interface ITwitterAccessor
    {
        Task<ITwitterResult> ExecuteRequestAsync(ITwitterRequest request);

        Task<ITwitterResult<T>> ExecuteRequestAsync<T>(ITwitterRequest request) // Example: T = IUploadInitModel
            where T : class;

        Task PrepareTwitterRequestAsync(ITwitterRequest request);
    }
}

File: WorldFeed/Client/src/app/core/Core/Web/ITwitterAccessor.ts
import {ITwitterRequest} from "../../Public/Models/Interfaces/ITwitterRequest";
import {ITwitterResult} from "./TwitterResult";

export interface ITwitterAccessor {
  executeRequestAsync(request: ITwitterRequest): Promise<ITwitterResult>;

  executeRequestAsync<T>(request: ITwitterRequest): Promise<ITwitterResult<T>>;

  prepareTwitterRequestAsync(request: ITwitterRequest): Promise<void>;
}
